graphql/src/store/query.rs

```diff
@@ -287,6 +289,39 @@ fn build_child_filter_from_object(
                 })
                 .collect(),
         ))
+    } else if entity.is_interface() {
+        Ok(EntityFilter::Or(
+            entity
+                .object_types(schema.schema())
+                .ok_or(QueryExecutionError::AbstractTypeError(
+                    "Interface is not implemented by any types".to_string(),
+                ))?
+                .iter()
+                .map(|object_type| {
+                    let field = object_type
+                        .fields
+                        .iter()
+                        .find(|f| f.name == field_name.clone())
+                        .ok_or(QueryExecutionError::InvalidFilterError)?;
+                    let derived = field.is_derived();
+
+                    let attr = match derived {
+                        true => sast::get_derived_from_field(child_entity, field)
+                            .ok_or(QueryExecutionError::InvalidFilterError)?
+                            .name
+                            .to_string(),
+                        false => field_name.clone(),
+                    };
+
+                    Ok(EntityFilter::Child(Child {
+                        attr: attr.clone(),
+                        entity_type: EntityType::new(child_entity.name().to_string()),
+                        filter: filter.clone(),
+                        derived,
+                    }))
+                })
+                .collect::<Result<Vec<EntityFilter>, QueryExecutionError>>()?,
+        ))
```

Comment:
Wow .. nice. That's a pretty nasty case. We'll see how it works out in practice, but it could be that all those `or`'s end up being slow and we need to change how the query is generated; but it's easier to determine that once we have an example of that.